src/main/java/com/gmail/nossr50/listeners/BlockListener.java

```diff
@@ -134,17 +135,23 @@ public void onBlockDropItemEvent(BlockDropItemEvent event) {
                     }
                 }
 
-                if (event.getBlock().getMetadata(MetadataConstants.METADATA_KEY_BONUS_DROPS).size()
-                        > 0) {
+                if (!event.getBlock().getMetadata(MetadataConstants.METADATA_KEY_BONUS_DROPS).isEmpty()) {
                     final BonusDropMeta bonusDropMeta =
                             (BonusDropMeta) event.getBlock().getMetadata(
                                     MetadataConstants.METADATA_KEY_BONUS_DROPS).get(0);
                     int bonusCount = bonusDropMeta.asInt();
                     final Location centeredLocation = getBlockCenter(event.getBlock());
-                    for (int i = 0; i < bonusCount; i++) {
+                    int newAmount = is.getAmount() * (bonusCount + 1);
 
-                        ItemUtils.spawnItemNaturally(event.getPlayer(),
-                                centeredLocation, is, ItemSpawnReason.BONUS_DROPS);
+                    ItemStack copy = is.clone();
+                    copy.setAmount(newAmount);
+
+                    final McMMOItemSpawnEvent spawnEvent = new McMMOItemSpawnEvent(centeredLocation,
```

Comment:
I'm not sure it makes sense for this event to fire if all we are doing is modifying the amount of the itemStack in this event. Instead, maybe we need a new event, something like McMMOAddBonusDrops, and it shows the items, the number of items that will be added, the final result, and is cancellable. If it goes uncancelled, or modified, the result affects the outcome of this listener.